solver/request.go

```diff
@@ -71,6 +71,7 @@ func (r *singleRequest) Solve(ctx context.Context, cln *client.Client, mw *Multi
 	})
 
 	g.Go(func() error {
+		defer s.Close()
```

Comment:
Were we just never closing sessions?

Reply:
BuildKit was automatically closing them on the first solve, which was broken: https://github.com/moby/buildkit/pull/2995

Reply:
There is a PR to buildkit linked in the description, it was always closed in the client.Solve even if it was passed in externally with the intent to use as a shared session.  WIth the buildkit fix we now need to close our own sessions since we are creating them outside the client.Solve.  This does give us the flexibility in the future to actually share and reuse sessions across multiple solves though.